### ISSUE TITLE
[PITR][Tablet Splitting]: Corruption (yb/util/uuid.cc:305): Invalid length of binary data with TxnSnapshotId  on executing PITR after a rolling restart , the tserver process goes down

### ISSUE THREAD

**Issue by kripasreenivasan** [yugabyte/yugabyte-db#14487]
Jira Link: [DB-3881](https://yugabyte.atlassian.net/browse/DB-3881)
### Description

- Create a cluster with the following gflags
Master:
```
tablet_split_low_phase_size_threshold_bytes= 13421
enable_automatic_tablet_splitting = true
tablet_split_low_phase_shard_count_per_node = 16
```
Tserver:
```
enable_automatic_tablet_splitting = true
ysql_num_shards_per_tserver = 1
yb_num_shards_per_tserver =1
```
- Create a colocated db inside this cluster
- Create 4 tables in the colocated db where the tables contain sequences(used share/schema.sql)
- Create a snapshot schedule with retention of 60 mins and frequency 5 mins
```bin/yb-admin -master_addresses 172.151.27.228,172.151.42.107,172.151.59.14 create_snapshot_schedule 5 60 ysql.cl_db```
- Create materialized views for all tables in the colocated db
- Insert data in the tables, refresh materialized views
-  Edit the cluster to enable YSQL packed rows OR execute a rolling restart on the cluster.
- Continue the workloads(DML + DDL)
- Once the gflag update is complete
- Execute the first point in time restore
**Issue:**
1. Upon executing this restore, the tserver process in 2/3 nodes crash
2. FATALS flood the tserver logs: 

```
F1017 07:27:51.369659 16865 operation_driver.cc:396] T d78ee487e2fc473c9bbe713c534b95c7 P 3e1802c5be404573b30d14a67a894a19 S RD-P Ts { physical: 1665991240419554 } kSnapshot (0x00000000069b2c60): Apply failed: Corruption (yb/util/uuid.cc:305): Invalid length of binary data with TxnSnapshotId '': 0 (expected 16), request: dest_uuid: "6b747ee5cf7f4e4c8ca182e4cc18a694" operation: RESTORE_ON_TABLET tablet_id: "d78ee487e2fc473c9bbe713c534b95c7" propagated_hybrid_time: 6823900120751927296 snapshot_hybrid_time: 6823897551530618880 restoration_id: "\210aK\275\3632E\007\201J~\343\343eud" schema { columns { id: 0 name: "db_oid" type { main: INT64 } is_key: true is_hash_key: true is_nullable: false is_static: false is_counter: false sorting_type: 0 order: 1936028263 pg_type_oid: 0 } columns { id: 1 name: "seq_oid" type { main: INT64 } is_key: true is_hash_key: true is_nullable: false is_static: false is_counter: false sorting_type: 0 order: 0 pg_type_oid: 0 } columns { id: 2 name: "last_value" type { main: INT64 } is_key: false is_nullable: false is_static: false is_counter: false sorting_type: 0 order: 0 pg_type_oid: 0 } columns { id: 3 name: "is_called" type { main: BOOL } is_key: false is_nullable: false is_static: false is_counter: false sorting_type: 0 order: 0 pg_type_oid: 0 } table_properties { contain_counters: false is_transactional: false consistency_level: STRONG use_mangled_column_name: false is_ysql_catalog_table: false retain_delete_markers: false partitioning_version: 1 } colocated_table_id { } pgschema_name: "" } db_oid: 16384
```
3. Core files generated

**Note:**
The Colocated DB does not affect this scenario. i retried this scenario without colocation and i was able to reproduce it.

CC: @sanketkedia , @lingamsandeep , @spolitov , @bmatican , @shamanthchandra-yb

**Comment by rthallamko3** [comment]
@sanketkedia , Is this manifestation same as the one you are fixing?

**Comment by kripasreenivasan** [comment]
This issue is reproducible without packed columns also. I substituted a rolling restart in place of the enabling packed rows step mentioned in the description.
CC: @bmatican

**Comment by kripasreenivasan** [comment]
@bmatican , @lingamsandeep , @sanketkedia 

I repeated this scenario without colocation and it is reproducible. i have updated the description. The colocated schema this issue was initially reproduced with does not have any bearing with the issue.

**Comment by kripasreenivasan** [comment]
This is not reproducible now. @lingamsandeep , Any idea which commit fixed this? This used to be consistently reproducible.

**Comment by agsh-yb** [comment]
Getting fatal with `2.14.6.0` while PITR restore 
Gflags:

-    master 
```
            "enable_automatic_tablet_splitting": "true",
            "tablet_split_high_phase_shard_count_per_node": 200,
            # high_phase_size 2MB
            "tablet_split_high_phase_size_threshold_bytes": 2097152,
            # low_phase_size 100KB
            "tablet_split_low_phase_size_threshold_bytes": 102400,
            "tablet_split_low_phase_shard_count_per_node": 16,
```

     
-  tserver 

```
            "enable_automatic_tablet_splitting": "true",
            "ysql_num_shards_per_tserver": 1,
```

Steps:
        1. Create universe with automatic splitting enabled, with RPC(algo=2)
        2. Create Databases(colocated) and Create colocated(50 tables) and Create 2
        complex tables(t1, t2) (1 Hashed 1 Range)
        3. Insert data to tables and validate data in tables
        4. Create Unique, Secondary and Partial indexes for both tables and colocated table.
        5. Create snapshot_schedule
        6. COPY TO colocated tables and non-colocated tables
        7. Collect time(1) and Drop colocated tables and validate
        8. Insert some data and validate
        9. PITR at time(1)
Failing at step 9 with fatals 

Attached logs files : [logs.zip](https://github.com/yugabyte/yugabyte-db/files/11190052/logs.zip)

**Comment by kripasreenivasan** [comment]
Ticket reopened since we are missing 2.14 backports.
CC: @lingamsandeep

**Comment by rthallamko3** [comment]
@sanketkedia , @yamen-haddad , Which fix is this issue referring to? Can we backport the fix to 2.14 branches as tablet splitting is GA in those branches?

**Comment by yamen-haddad** [comment]
This issue refers to the following two fixes:

-  [D19631](https://phabricator.dev.yugabyte.com/D19631) It was fixed by Mingda but hadn't been backported to 2.14. I am working on the backport and should submit the diff for review tonight (had some annoying merge conflicts).

- [D23857](https://phabricator.dev.yugabyte.com/D23857). This is the fix about sequence + tablet splitting. I have already backported it to 2.14.
